django/library/serializers.py

```diff
@@ -110,19 +190,46 @@ def get_profile_url(self, instance):
             )
 
     def update(self, instance, validated_data):
-        affiliations = validated_data.pop("affiliations", None)
-        validated_data.pop("given_name", None)
-        validated_data.pop("family_name", None)
+        if not self._is_not_used_or_used_by_current_release_only(instance):
+            if self._trying_to_modify_attributes(instance, validated_data):
```

Comment:
does this update() get called when we do stuff like re-arranging/changing roles? I'm not sure what the `is_trying_to_modify..` check is for

Reply:
yeah, `update()` is called when contributors are rearranged and roles are changed too. 
If a contributor is already "used" for another codebase, we still need to be able to add him to any other codebase, just need to make sure that the PUT request (which is called on that newly added contributor) doesn't modify any of it's attributes. Hence the naming: `_trying_to_modify_attributes()`

we are trying to squeeze to much functionality through one route `/contributors`...
Next improvement could be to split CREATE and UPDATE actions into 2 different routes in DRF.

Reply:
wonder if its possible to change something in the ReleaseContributorSerializer so that `update()` on the contributor serializer doesn't get invoked when we are just changing the index/roles/etc

This probably works, the method to check whether we are updating in the update() method just felt wrong lol